### PR TITLE
[Core] `aaz`: Add argument completer support

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

**Description**

Add completer support in aaz core.

Customer could use custom completer:

```Python
from azure.cli.core.decorators import Completer

@Completer
def _get_supported_languages(cmd, prefix, namespace):
    return ["en-us", "es-es", "fr-fr", "de-de"]

class FooCreate(_CreateFoo):

    @classmethod
    def _build_arguments_schema(cls, *args, **kwargs):
        if cls._args_schema is not None:
            return cls._args_schema
        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)

        _args_schema = cls._args_schema
        _args_schema.bar._completer = _get_supported_languages

        return cls._args_schema
```

image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

**Comment by kairu-ms** [comment]
Please fix those lines as well 
- https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L508
- https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L549
- https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L583

**Comment by ReaNAiveD** [comment]
> Please fix those lines as well
> 
> * https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L508
> * https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L549
> * https://github.com/Azure/azure-cli/blob/fc8e7e9267509706293a53929cc472ca18dd11a3/src/azure-cli-core/azure/cli/core/aaz/_arg.py#L583

Done

File: src/azure-cli-core/azure/cli/core/aaz/_arg.py [hunk 1]
```diff
@@ -498,7 +505,8 @@ def to_cmd_arg(self, name, **kwargs):
         from azure.cli.core.commands.parameters import get_resource_group_completion_list
         from azure.cli.core.local_context import LocalContextAttribute, LocalContextAction, ALL
         arg = super().to_cmd_arg(name, **kwargs)
-        arg.completer = get_resource_group_completion_list
+        if not arg.completer:
+            arg.completer = get_resource_group_completion_list
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/aaz/_arg.py:509]: you can set completer in __init__ function

File: src/azure-cli-core/azure/cli/core/aaz/_arg.py [hunk 2]
```diff
@@ -562,18 +574,20 @@ def __init__(
...
         fmt = fmt or AAZSubscriptionIdArgFormat()
+        completer = completer or get_subscription_id_list
         super().__init__(
             help=help,
             fmt=fmt,
+            completer=completer,
             **kwargs
         )
 
     def to_cmd_arg(self, name, **kwargs):
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/aaz/_arg.py:589]: Can clean up this method overwritten.

```suggestion
    def to_cmd_arg(self, name, **kwargs):
```

Comment (ReaNAiveD) [reply src/azure-cli-core/azure/cli/core/aaz/_arg.py]: Done

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**